### PR TITLE
SDK: Add ts sdk

### PR DESCRIPTION
**Pull request by barnjamin**
Add Ntt TS SDK
--------------

Definition of interface and layouts in `/sdk/definitons`
Evm and Solana implementations in `/sdk/evm|solana`

Replaces `ci_tests` directory, tests ported to use SDK and moved to `sdk/__tests__`

Replaces `anchor` Solana test with `solana/__tests__/ntt.test.ts` and modify Makefile to build bindings && run test from the context of the Solana SDK

### CODE REVIEW

File: sdk/evm/src/ntt.ts [hunk 1]
```diff
@@ -160,18 +162,24 @@ export class EvmNtt<N extends Network, C extends EvmChains>
...
     amount: bigint,
     destination: ChainAddress,
-    queue: boolean
+    queue: boolean,
+    relay?: boolean
   ): AsyncGenerator<EvmUnsignedTransaction<N, C>> {
-    const [_, totalPrice] = await this.quoteDeliveryPrice(destination.chain);
+    let totalPrice = 0n;
+    if (relay) {
+      [, totalPrice] = await this.quoteDeliveryPrice(destination.chain);
```
Comment (kev1n-peters) [inline sdk/evm/src/ntt.ts:178]: `quoteDeliveryPrice` will return the wormhole fee (currently `0`) for manual transfers. I think we should call this even if `relay === false`. We will need to pass the `relay` flag to `quoteDeliveryPrice` and encode it

File: sdk/evm/__tests__/versions.test.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { Wormhole } from "@wormhole-foundation/sdk-connect";
+import { EvmPlatform } from "@wormhole-foundation/sdk-evm";
+import { EvmNtt } from "../src/index.js";
+
+const wh = new Wormhole("Testnet", [EvmPlatform]);
+
+const overrides = {
+  Avalanche: {
+    token: "0x72CAaa7e9889E0a63e016748179b43911A3ec9e5",
```
Comment (artursapek) [inline sdk/evm/__tests__/versions.test.ts:9]: this doesnt exist!!!!!!!!

File: sdk/solana/src/ntt.ts [hunk 3]
```diff
@@ -0,0 +1,805 @@
...
+  async getConfig(): Promise<Config> {
+    this.config =
+      this.config ??
+      (await this.program.account.config.fetch(this.pdas.configAccount()));
+    return this.config;
+  }
+
+  async getTokenDecimals(): Promise<number> {
+    // TODO:
+    return 9;
```
Comment (kev1n-peters) [inline sdk/solana/src/ntt.ts:148]: We can get the `mint` address from the `config` and then fetch the token metadata for the decimals https://github.com/wormhole-foundation/example-native-token-transfers/blob/1193a4a00a8b32aeb0f8705913f09f9bfc3e53c2/solana/programs/example-native-token-transfers/src/config.rs#L17

File: sdk/evm/__tests__/versions.test.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    manager: "0x22D00F8aCcC2da440c937104BA49AfD8261a660F",
+    transceiver: {
+      wormhole: "0xeA8D34fa9147863e486d2d07AB92b8218CF58C0E",
+    },
+  },
+};
+
+describe("ABI Versions Test", function () {
```
Comment (kev1n-peters) [inline sdk/evm/__tests__/versions.test.ts:17]: Solana now has a version (example of how to fetch is in this PR https://github.com/wormhole-foundation/example-native-token-transfers/pull/400). Should we add a version check to the Ntt Solana implementation?

File: sdk/solana/src/ntt.ts (hunk 3 above)
Comment (barnjamin) [reply sdk/solana/src/ntt.ts]: added this

File: sdk/evm/__tests__/versions.test.ts (hunk 4 above)
Comment (barnjamin) [reply sdk/evm/__tests__/versions.test.ts:17]: added in https://github.com/wormhole-foundation/example-native-token-transfers/pull/407

**Review by mdulin2 — APPROVED** [review]
The CI Tests look much nicer now than when I originally wrote them. If the CI still works then I'm happy.

**Review by kev1n-peters — APPROVED** [review]

**Pull request merged**